### DISCUSSION TITLE
Using `a17-mediafield` in a custom page

### DISCUSSION THREAD

**Discussion by dacastro4** [area17/twill#889] [Q&A]
I'm trying to use this `a17-mediafield` vue component in a custom page so the user is able to upload an image and attach it to a module but I can't pre select o set an image as selected from the component only from the vuex store which is not even available from the custom pages.

I would appreciate some help here.

This is what is on my custom page

```
<a17-inputframe label="Media" name="medias.mainImage">
    <a17-mediafield
        name="mainImage"
        crop-context="mainImage"
        :max="1"
        :with-add-info="false"
        :with-video-url="false"
        :with-caption="false"
        :alt-text-max-length="200"
        :caption-max-length="200"
        :button-on-top="true"
    >
    </a17-mediafield>
</a17-inputframe>
```

**Comment by pboivin** [comment]
Hi @dacastro4,

If your goal is only to attach an image to a model, the easiest way would be to use `@formField('medias', [ ... ])` in your module's `form.blade.php`.

If you wish to use `<a17-mediafield>` in the context of a custom CMS page, you need to implement the field's save/prefill logic inside of your custom page controller and view.

Would you be able to share a bit more details (or code) on what you are trying to accomplish in your custom page? I'm interested :)
